apps/go-mod-validator/src/index.ts

```diff
@@ -0,0 +1,90 @@
+import { validateDependency } from "./github";
+import { getDependenciesMap } from "./deps";
+import * as core from "@actions/core";
+import minimist from "minimist";
+import { Octokit } from "octokit";
+
+const smartContractKitPrefix = "github.com/smartcontractkit";
+
+function getContext() {
+  const argv = minimist(process.argv.slice(2));
+  const { local, tokenEnv, goModDir } = argv;
+
+  const dir = local ? goModDir || `"$(pwd)"` : core.getInput("go-mod-dir");
+
+  const githubToken = local
+    ? process.env[tokenEnv] || ""
+    : core.getInput("github-token");
+
+  return { goModDir: dir, octokitClient: new Octokit({ auth: githubToken }) };
+}
+
+async function run() {
+  const { goModDir, octokitClient } = getContext();
+
+  // get dependencies from go.mod file
+  let dependenciesMap: Map<string, any> = new Map();
+  try {
+    dependenciesMap = getDependenciesMap(goModDir);
+  } catch (err) {
+    core.info(`failed to get dependencies, err: ${err}`);
+    core.setFailed(`failed to get dependencies`);
+  }
+
+  // Verify each of the dependencies
+  const validationFailedDependencies: string[] = [];
+  for (const [file, dependencies] of dependenciesMap.entries()) {
+    for (let dependency of dependencies) {
+      // handle replace redirectives
+      if (dependency.Replace != undefined) {
+        dependency = dependency.Replace;
+      }
+
+      // `go list -m -json all` also lists the main pacakge, avoid parsing it.
+      // and only validate dependencies belonging to our org
+      if (
+        dependency.Version == undefined ||
+        !dependency.Path.startsWith(smartContractKitPrefix)
+      ) {
+        continue;
+      }
+
+      // prepare dependency result string
+      let dependencyResult = `${dependency.Path}@${dependency.Version}`;
+      if (dependency.Indirect == true) {
+        dependencyResult += " // indirect";
+      }
+
+      // validate the dependency
+      try {
+        if (
+          !(await validateDependency(
+            dependency.Path,
+            dependency.Version,
+            octokitClient,
+          ))
+        ) {
+          validationFailedDependencies.push(dependencyResult);
+        }
+      } catch (err) {
+        core.info(
+          `failed to verify dependency: ${dependency.Path}@${dependency.Version}, err: ${err}`,
+        );
+        validationFailedDependencies.push(dependencyResult);
+      }
+    }
+  }
+
+  if (validationFailedDependencies.length != 0) {
+    core.info(
+      `validation failed for following dependencies:\n${validationFailedDependencies.join("\n")}`,
+    );
+    core.setFailed(
+      `validation failed for ${validationFailedDependencies.length} dependencies`,
+    );
```

Comment:
I mentioned this in the last review. A developer who updates the go.mod might not know anything about this workflow, and these error messages are not enough to know what went wrong, and how to fix them. 

Some questions that should probably be answered:
- What validation failed?
- Where is the error?
- How can I fix it?

Reply:
My bad, I've fixed it now.
- Errors are printed along with failed dependencies
- Also I've put in how to fix failed dependencies errors in the README and added a message in `core.setFailed` to read to README

Tested here:
- Validation failed - [chainlink-data-streams](https://github.com/smartcontractkit/chainlink-data-streams/actions/runs/9898867493)
- Validation failed - [chainlink](https://github.com/smartcontractkit/chainlink/actions/runs/9898891582)
- Validation successful - [crib](https://github.com/smartcontractkit/crib/actions/runs/9898871252)